viewing_party/party.py

```diff
@@ -1,23 +1,168 @@
+    friend_data = get_friend_data(user_data)
+
+    #iterate over friend data and if movie isn't in friend data, add it to unique_watched
+    for i in range(len(user_data["watched"])):
+        if user_data["watched"][i] not in friend_data:
+            unique_watched.append(user_data["watched"][i])
 
-        
+    return unique_watched
+
+def get_friends_unique_watched(user_data):
+    friends_unique_watched = []
+
+    #iterate over friends' watched lists, and if it's not in the user's list,
+    #append it to a new list and return it
+    for i in range(len(user_data["friends"])):
+        for j in range(len(user_data["friends"][i]["watched"])):
+            if user_data["friends"][i]["watched"][j] not in user_data["watched"]:
+                if user_data["friends"][i]["watched"][j] not in friends_unique_watched:
+                    friends_unique_watched.append(user_data["friends"][i]["watched"][j])
+    return friends_unique_watched
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    #initialize data
+    recommendations = []
+    friends_unique_watched = get_friends_unique_watched(user_data)
 
+    #go through list of subscriptions, then find which among user's friend's unique watched
+    #user hasn't watched
+    for i in range(len(user_data["subscriptions"])):
+        for j in range(len(friends_unique_watched)):
+            if user_data["subscriptions"][i] == friends_unique_watched[j]["host"]:
+                recommendations.append(friends_unique_watched[j])
+    return recommendations
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    #initialize data
+    friends_unique_watched = get_friends_unique_watched(user_data)
+    most_frequent_genre = get_most_watched_genre(user_data)
+    recommendations = []   
+    #interate over the movies that user's friends have watched,
+    #that user has not watched
+    for i in range(len(friends_unique_watched)):
+        #if a friend's movie is user's favorite genre, recommend it
+        if friends_unique_watched[i]["genre"] == most_frequent_genre:
+            recommendations.append(friends_unique_watched[i])
+    return recommendations
+
+def get_rec_from_favorites(user_data):
+    recommendations = []
+
+    #get list of all friend movies watched
+    friend_data = get_friend_data(user_data)
+
+    #interate over list of favorites, then see what's in favorites 
+    #that aren't in friends watched list.
+    for i in range(len(user_data["favorites"])):
+        for j in range(len(friend_data)):
+            if user_data["favorites"][i] not in friend_data:
+                if user_data["favorites"][i] not in recommendations:
+                        recommendations.append(user_data["favorites"][i])
+    return recommendations
```

Comment:
If we called `get_unique_watched(user_data)` instead, we could simplify the loops below a little:
```suggestion
    unique_watched = get_unique_watched(user_data)
    
    for movie in unique_watched:
        if movie in user_data["favorites"]:
               recommendations.append(movie)

    return recommendations
```